lib/encoder.js

```diff
@@ -0,0 +1,35 @@
+const abi = require('web3-eth-abi')
+const FINANCE_ABI = require('../artifacts/Finance.json').abi
+const FORWARDER_ABI = require('../artifacts/IForwarderWithContext.json').abi
+
+const CALLSCRIPT_ID = '0x00000001'
+
+function encodeCallsScript(actions) {
+  return actions.reduce((script, { to, data }) => {
+    const address = abi.encodeParameter('address', to)
+    const dataLength = abi.encodeParameter('uint256', (data.length - 2) / 2).toString('hex')
+    return script + address.slice(26) + dataLength.slice(58) + data.slice(2)
```

Comment:
This is probably `callscript` specific :) But what are these two lines doing, exactly?

Reply:
`dataLength` basically is the size of the calldata that will be encoded in the callscript encoded to the size of an EVM word (32 bytes).

The second line is basically removing the leading zeros and `0x` from the variables encoded above. For example, addresses are 20 bytes long, but since the encoded `address` is 33 bytes long (`0x` + 32 bytes), we remove 26 leading nibbles.